api/handlers.go

```diff
@@ -0,0 +1,382 @@
+	if err != nil {
+		http.Error(w, `{"error":"Неверный идентификатор"}`, http.StatusBadRequest)
+		return
+	}
+
+	task, err := database.FetchTaskByID(id) // Получаем задачу из базы данных по ID
+	if err != nil {
+		http.Error(w, fmt.Sprintf(`{"error":"%s"}`, err.Error()), http.StatusNotFound)
+		return
+	}
+
+	response := map[string]interface{}{
+		"id":      task.ID,
+		"date":    task.Date,
+		"title":   task.Title,
+		"comment": task.Comment,
+		"repeat":  task.Repeat,
+	}
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	err = json.NewEncoder(w).Encode(response)
+	if err != nil {
+		http.Error(w, `{"error":"Ошибка форматирование JSON-ответа"}`, http.StatusInternalServerError)
+		return
+	}
+}
+
+// PutTaskHandler обновляет задачу
+func PutTaskHandler(w http.ResponseWriter, r *http.Request) {
+	log.Println("Получен PUT-запрос для обновления задачи")
+
+	if r.Method != http.MethodPut {
+		log.Println("Метод не поддерживается")
+		http.Error(w, `{"error":"Метод не поддерживается"}`, http.StatusMethodNotAllowed)
+		return
+	}
+	// Создаем переменную для задачи
+	var task schemas.Table
+	if err := json.NewDecoder(r.Body).Decode(&task); err != nil {
+		log.Println("Ошибка декодирования JSON:", err)
+		http.Error(w, `{"error":"Ошибка декодирования JSON"}`, http.StatusBadRequest)
+		return
+	}
+
+	if task.ID == "" {
+		log.Println("Не указан идентификатор задачи")
+		http.Error(w, `{"error":"Не указан идентификатор задачи"}`, http.StatusBadRequest)
+		return
+	}
+
+	if task.Title == "" {
+		log.Println("Не указан заголовок задачи")
+		http.Error(w, `{"error":"Не указан заголовок задачи"}`, http.StatusBadRequest)
+		return
+	}
+
+	now := time.Now()
+	today := now.Format("20060102")
+	if task.Date == "" || task.Date == "today" {
+		task.Date = today
+	} else {
+		parsedDate, err := time.Parse("20060102", task.Date) // Парсим дату задачи
+		if err != nil {
+			log.Println("Некорректный формат даты:", err)
+			http.Error(w, `{"error":"Некорректный формат даты"}`, http.StatusBadRequest)
+			return
+		}
+		if parsedDate.Before(now) {
+			if task.Repeat == "" {
+				task.Repeat = today // Устанавливаем дату на сегодня
+			} else {
+				nextDate, err := rules.NextDate(now, task.Date, task.Repeat)
+				if err != nil {
+					log.Println("Некорректное правило повторения:", err)
+					http.Error(w, `{"error":"Некорректное правило повторения"}`, http.StatusBadRequest)
+					return
+				}
+				if task.Date != today {
+					task.Date = nextDate // Устанавливаем следующую дату
+				}
+			}
+		}
+	}
+	if task.Repeat != "" {
+		if _, err := rules.NextDate(now, task.Date, task.Repeat); err != nil {
+			log.Println("Некорректное правило повторения:", err)
+			http.Error(w, `{"error":"Некорректное правило повторения"}`, http.StatusBadRequest)
+			return
+		}
+	}
+	// Обновление задачи в базе данных
+	if err := database.UpdateTask(task); err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
Для сравнивания ошибок нужно использовать специализированную функцию
```suggestion
		if errors.Is(err, sql.ErrNoRows) {
```